kandinsky/src/context_rect.cpp

```diff
@@ -14,7 +14,7 @@ void KDContext::fillRect(KDRect rect, KDColor color) {
 }
 
 /* Note: we support the case where workingBuffer IS equal to pixels */
-void KDContext::fillRectWithPixels(KDRect rect, const KDColor * pixels, KDColor * workingBuffer) {
+void KDContext::fillRectWithPixels(KDRect rect, const KDColor * pixels, KDColor * workingBuffer, KDColor * prevPixels) {
```

Comment:
I'm rather uncomfortable modifying this, and I also have a hard time figuring out what you're trying to achieve here 😄 Would you mind explaining so we can see if there's a better solution?

Reply:
In order to move the turtle across the screen, I need to erase the turtle icon by blitting back the pixels under it, so I need a way to fetch those pixels that's more efficient than brute-forcing `getPixel()`. This is a very dirty hack to write-back the pixels under the turtle, but I've changed it to a more conventional `fetchPixels(KDRect rect, KDColor * pixels)` in my current Python rewriting (not online yet).

Reply:
I see 😄I think such an API could make sense in `KDContext` (a wrapper around `pullRect` and `absoluteFillRect`).